chain/chain/src/resharding/manager.rs

```diff
@@ -250,6 +272,108 @@ impl ReshardingManager {
+        new_shard_uid: ShardUId,
+        parent_shard_uid: ShardUId,
+        retain_mode: RetainMode,
+    ) -> Result<ChunkExtra, Error> {
+        let mut child_chunk_extra = ChunkExtra::clone(parent_chunk_extra);
+        *child_chunk_extra.state_root_mut() = new_state_root;
+
+        if let Some(congestion_info) = child_chunk_extra.congestion_info_mut() {
+            let &parent_state_root = parent_chunk_extra.state_root();
+            *congestion_info = self.get_child_congestion_info(
+                block,
+                tries,
+                parent_shard_uid,
+                parent_state_root,
+                new_shard_uid,
+                new_state_root,
+                retain_mode,
+                &congestion_info,
+            )?;
+
+            // Please note the usage of the child shard layout here.
+            let next_epoch_id = self.epoch_manager.get_next_epoch_id(block.hash())?;
+            let next_shard_layout = self.epoch_manager.get_shard_layout(&next_epoch_id)?;
+            let all_shards = next_shard_layout.shard_ids().collect_vec();
+            let own_shard = new_shard_uid.shard_id();
+            let own_shard_index = next_shard_layout
+                .get_shard_index(own_shard)?
+                .try_into()
+                .expect("ShardIndex must fit in u64");
+
+            // Use simplified congestion seed. The proper one should be
+            // block height + shard index, however the block heigh is not
+            // easily available in all required places.
+            let congestion_seed = own_shard_index;
+            congestion_info.finalize_allowed_shard(own_shard, &all_shards, congestion_seed);
+        }
+        Ok(child_chunk_extra)
+    }
+
+    fn get_child_congestion_info(
+        &mut self,
+        block: &Block,
+        tries: &ShardTries,
+        parent_shard_uid: ShardUId,
+        parent_state_root: CryptoHash,
+        new_shard_uid: ShardUId,
+        new_state_root: CryptoHash,
+        retain_mode: RetainMode,
+        congestion_info: &CongestionInfo,
+    ) -> Result<CongestionInfo, Error> {
+        if retain_mode == RetainMode::Left {
+            return Ok(congestion_info.clone());
+        }
+
+        // left child -> unchanged
+        // right child -> remove the buffered receipts
+        let epoch_id = block.header().epoch_id();
+        let shard_layout = self.epoch_manager.get_shard_layout(&epoch_id)?;
+        let protocol_version = self.epoch_manager.get_epoch_protocol_version(epoch_id)?;
+
+        let trie = tries.get_trie_for_shard(parent_shard_uid, parent_state_root);
+        let mut smart_congestion_info = congestion_info.clone();
+        for shard_id in shard_layout.shard_ids() {
+            let receipt_groups = ReceiptGroupsQueue::load(&trie, shard_id)?;
+            let Some(receipt_groups) = receipt_groups else {
+                tracing::info!(target: "boom", ?shard_id, "no receipt group found!");
+                continue;
+            };
```

Comment:
@jancionear I'm getting "no receipt group found" here for all shards, can you have a quick look if I'm doing anything obviously wrong? 

The "brute force" solution now works and there should be non-zero receipts in the parent shard's trie.

Reply:
I see that `resharing_v3` runs at protocol version for `SimpleNightshadeV4`, bandwidth scheduler probably isn't enabled in the test:

```
ProtocolFeature::SimpleNightshadeV4 => 145,
#[cfg(feature = "protocol_feature_relaxed_chunk_validation")]
ProtocolFeature::RelaxedChunkValidation => 146,
ProtocolFeature::ExcludeExistingCodeFromWitnessForCodeLen => 147,
ProtocolFeature::BandwidthScheduler => 148,
```

Moving `BandwidthScheduler` before `SimpleNightshadeV4` should help

Reply:
Oh my, you're very close, I think the test goes all the way to nightly but we need to have `BandwidthScheduler` enabled **before** resharding. Thanks!